.werft/platform-delete-preview-environments-cron.ts

```diff
@@ -37,38 +39,92 @@ async function deletePreviewEnvironments() {
     }
     werft.done("prep")
 
-    werft.phase("Fetching branches");
-    const branches = getAllBranches();
-    // During the transition from the old preview names to the new ones we have to check for the existence of both the old or new
-    // preview name patterns before it is safe to delete a namespace.
-    const expectedPreviewEnvironmentNamespaces = new Set(branches.flatMap(branch => [parseBranch(branch), expectedNamespaceFromBranch(branch)]));
-    werft.done("Fetching branches");
-
     werft.phase("Fetching previews");
     let previews: string[]
     try {
         previews = listAllPreviewNamespaces(CORE_DEV_KUBECONFIG_PATH, {});
-        previews.forEach(previewNs => werft.log("Fetching previews", previewNs))
     } catch (err) {
         werft.fail("Fetching previews", err)
     }
     werft.done("Fetching previews");
 
+    werft.phase("Fetching outdated branches");
+    const branches = getAllBranches();
+    const outdatedPreviews = [];
+    // During the transition from the old preview names to the new ones we have to check for the existence of both the old or new
+    // preview name patterns before it is safe to delete a namespace.
+    branches.forEach(branch => {
+        const lastCommit = exec(`git log ${branch} --since=$(date +%Y-%m-%d -d "5 days ago")`, { silent: true })
+        if ( lastCommit.length < 1 ) {
+            const outdatedPreview = parseBranch(branch)
+            outdatedPreviews.push(outdatedPreview.trim())
+        }
+    })
+    const expectedPreviewEnvironmentNamespaces = new Set(branches.flatMap(branch => [parseBranch(branch), expectedNamespaceFromBranch(branch)]));
+
+    werft.phase("checking activity")
+    previews.forEach(previewNS => {
+        checkActivity(previewNS);
+    })
+
     werft.phase("deleting previews")
     try {
-        const previewsToDelete = previews.filter(ns => !expectedPreviewEnvironmentNamespaces.has(ns))
+        previewsToDelete = previewsToDelete.concat(previews.filter(ns => !expectedPreviewEnvironmentNamespaces.has(ns)))
```

Comment:
Instead of mutating a global variable here I'd prefer to use local variables with meaningful names and then creating a Set in the end (to ensure we don't try to delete the same preview env more than once)

```typescript
const deleteDueToMissingNamespace = previews.filter(ns => !expectedPreviewEnvironmentNamespaces.has(ns))
const deleteDueToNoCommitActivity = previews.filter(ns => outdatedPreviews.has(ns))
const deleteDueToNoDBActivity = previews.filter(ns => checkActivity(ns))
const previewsToDelete = new Set(deleteDueToMissingNamespace, deleteDueToNoCommitActivity, deleteDueToNoDBActivity)
```

This assume that checkActivity return a bool given a namespace instead of mutating a global variable. See comment for checkActivity below for details.